mission_control/src/behaviors/waypoint.cpp

```diff
@@ -34,222 +34,116 @@
 
 // Original version: Christopher Scianna Christopher.Scianna@us.QinetiQ.com
 #include <string>
+
+#include <geodesy/utm.h>
+
 #include "mission_control/behaviors/waypoint.h"
 
-using mission_control::GoToWaypoint;
+namespace mission_control
+{
 
 GoToWaypoint::GoToWaypoint(const std::string& name, const BT::NodeConfiguration& config)
-    : Behavior(name, config)
+    : BT::ActionNodeBase(name, config)
 {
-  getInput<double>("depth", depth_);
-  if (depth_ != 0.0) depthEnable_ = true;
-
-  getInput<double>("altitude", altitude_);
-  if (altitude_ != 0.0) altitudeEnable_ = true;
-
-  getInput<double>("latitude", lat_);
-  if (lat_ != 0.0) latEnable_ = true;
-
-  getInput<double>("longitude", long_);
-  if (long_ != 0.0) longEnable_ = true;
-
-  getInput<double>("wp_radius", wpRadius_);
-  if (wpRadius_ != 0.0) wpRadiusEnable_ = true;
-
-  getInput<double>("speed_knots", speedKnots_);
-  if (speedKnots_ != 0.0) speedKnotsEnable_ = true;
-
-  getInput<double>("time_out", timeOut_);
-  subStateData_ =
-      nodeHandle_.subscribe("/pose/corrected_data", 1, &GoToWaypoint::stateDataCallback, this);
-
-  goalHasBeenPublished_ = false;
-  waypointBehaviorPub_ = nodeHandle_.advertise<mission_control::Waypoint>("/mngr/waypoint", 1);
-}
+  if (!getInput<double>("latitude", latitude_) || !getInput<double>("longitude", longitude_))
+  {
+    throw std::runtime_error("Either latitude or longitude were left unspecified");
```

Comment:
should we throw BT::RuntimeError instead?

Reply:
Good idea. See https://github.com/ChrisScianna/ROS-Underwater-RnD/pull/118/commits/1c6eb422880713214455b79b6a5e3b7fb38f4d8a.